code/game/objects/obj_defense.dm

```diff
@@ -174,6 +174,8 @@ GLOBAL_DATUM_INIT(acid_overlay, /mutable_appearance, mutable_appearance('icons/e
 
 ///the obj is deconstructed into pieces, whether through careful disassembly or when destroyed.
 /obj/proc/deconstruct(disassembled = TRUE)
+	if(atom_storage)
+		atom_storage.remove_all()
 	SEND_SIGNAL(src, COMSIG_OBJ_DECONSTRUCT, disassembled)
```

Comment:
why not just use this signal

Reply:
![image](https://user-images.githubusercontent.com/88991542/171567874-55a1fd9c-daac-48ce-9026-91bf50f059a1.png)

Reply:
they're better here even, since we're basically just reacting to an event.
the dumbshit is signals that aren't events, and are instead just direct "commands" to the component
stuff like try insert, return existing, or remove, etc. most of the storage signals really

the generic ones are fine